app/client/cli/helpers/common.go

```diff
@@ -10,5 +19,44 @@ var (
 	genesisPath = runtime.GetEnv("GENESIS_PATH", "build/config/genesis.json")
 
 	// P2PMod is initialized in order to broadcast a message to the local network
+	// TECHDEBT: prefer to retrieve P2P module from the bus instead.
 	P2PMod modules.P2PModule
 )
+
+// fetchPeerstore retrieves the providers from the CLI context and uses them to retrieve the address book for the current height
+func FetchPeerstore(cmd *cobra.Command) (types.Peerstore, error) {
+	bus, err := GetBusFromCmd(cmd)
+	if err != nil {
+		return nil, err
+	}
+	// TECHDEBT(#810, #811): use `bus.GetPeerstoreProvider()` after peerstore provider
+	// is retrievable as a proper submodule
+	pstoreProvider, err := bus.GetModulesRegistry().GetModule(peerstore_provider.PeerstoreProviderSubmoduleName)
+	if err != nil {
+		return nil, errors.New("retrieving peerstore provider")
+	}
+	currentHeightProvider := bus.GetCurrentHeightProvider()
+	height := currentHeightProvider.CurrentHeight()
+	pstore, err := pstoreProvider.(peerstore_provider.PeerstoreProvider).GetStakedPeerstoreAtHeight(height)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving peerstore at height %d", height)
+	}
+	// Inform the client's main P2P that a the blockchain is at a new height so it can, if needed, update its view of the validator set
+	err = sendConsensusNewHeightEventToP2PModule(height, bus)
+	if err != nil {
+		return nil, errors.New("sending consensus new height event")
+	}
+	return pstore, nil
+}
+
+// sendConsensusNewHeightEventToP2PModule mimicks the consensus module sending a ConsensusNewHeightEvent to the p2p module
+// This is necessary because the debug client is not a validator and has no consensus module but it has to update the peerstore
+// depending on the changes in the validator set.
+// TODO(#613): Make the debug client mimic a full node.
+func sendConsensusNewHeightEventToP2PModule(height uint64, bus modules.Bus) error {
```

Comment:
Let's either add a TECHDEBT comment or validate in this PR if it's necessary.

I prefer the former since ei might require some time to test